### PR TITLE
Security fixes

### PR DESCRIPTION
**Pull request by ptondereau**
This PR provides security hardening in PHP code.

### CODE REVIEW

File: src/Mollie/API/Client.php [hunk 1]
```diff
@@ -396,7 +401,7 @@ public function performHttpCall ($http_method, $api_method, $http_body = NULL)
 			 */
 			$request_headers[] = "X-Mollie-Debug: old OpenSSL found";
 			curl_setopt($this->ch, CURLOPT_HTTPHEADER, $request_headers);
-			curl_setopt($this->ch, CURLOPT_SSL_VERIFYHOST, 0);
+			curl_setopt($this->ch, CURLOPT_SSL_VERIFYHOST, 2);
```
Comment (ptondereau) [inline src/Mollie/API/Client.php]: Prevent MITM attack

Comment (willemstuursma) [inline src/Mollie/API/Client.php]: Thanks. I guess it is time to get rid of this failover.

File: src/Mollie/API/Autoloader.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ public static function autoload ($class_name)
 		if (strpos($class_name, "Mollie_") === 0)
 		{
 			$file_name = str_replace("_", "/", $class_name);
-			$file_name = realpath(dirname(__FILE__) . "/../../{$file_name}.php");
```
Comment (willemstuursma) [inline src/Mollie/API/Autoloader.php]: This won't work on some old PHP versions, sorry. We still need to support them.

File: src/Mollie/API/Client.php [hunk 3]
```diff
@@ -216,7 +221,7 @@ public function __construct ()
 		$this->addVersionString($curl_version["ssl_version"]);
 
 		// The PEM path may be overwritten with setPemPath().
-		$this->pem_path = realpath(dirname(__FILE__) . "/cacert.pem");
```
Comment (willemstuursma) [inline src/Mollie/API/Client.php]: This won't work on some old PHP versions, sorry. We still need to support them.

**Comment by willemstuursma** [comment]
Hi @ptondereau. I've removed the support for OpenSSL 0.9.7 in e0a4fd38f462b736462b613990cfaab31d131272. 

I'm afraid we won't be making the other changes, sorry.

Thanks for pointing this out and contributing to our module!

**Comment by it-can** [comment]
But what about the strict checks? Seems reasonable?

**Comment by ptondereau** [comment]
@it-can always do strict checks

**Comment by it-can** [comment]
In this PR there are a few strict checks added... maybe a good addition to the Mollie PHP lib

**Pull request closed without merging**